motionlayout/src/main/java/com/google/androidstudio/motionlayoutexample/fragmentsdemo/FragmentExample2Activity.kt

```diff
@@ -35,7 +35,7 @@ class FragmentExample2Activity : AppCompatActivity(), View.OnClickListener, Moti
                         .setCustomAnimations(R.animator.show, 0)
                 fragment = ListFragment.newInstance()
                 transaction
-                        .replace(R.id.container, fragment)
+                        .replace(R.id.container, fragment as ListFragment)
```

Comment:
Variable fragment is nullable but the replace method expects a non nullable fragment.
I think it might be the case that the @NonNull is added in a recent version of support library.
I have used (embarrassingly intermittently) explicit type cast and double exclamation to handle it. 
I think double exclamation is the better solution here and should be used in all cases uniformly.
Do you agree? Is there a third option?

Reply:
I see. I think using a scope function in the same line as the constructor makes it more readable.
Like this https://github.com/googlesamples/android-ConstraintLayoutExamples/pull/43/files#diff-9a9f2a1ceb3ead1c92fa955a508b8fe5R78

(Sorry for the delayed reply..)

Reply:
Yes that works. I've implemented and pushed it.